lib/_http_outgoing.js

```diff
@@ -125,6 +127,19 @@ function OutgoingMessage() {
 }
 util.inherits(OutgoingMessage, Stream);
 
+Object.defineProperty(OutgoingMessage.prototype, 'socket', {
+  get: function() {
+    return this[kSocket];
+  },
+  set: function(socket) {
+    this[kSocket] = socket;
+    if (socket && socket[async_id_symbol] === undefined) {
+      socket[async_id_symbol] = async_hooks.newUid();
```

Comment:
So as we wrote in https://github.com/nodejs/node/issues/14386#issuecomment-316674741
This _might_ not be enough. Is the socket is being reused, then it might hold a `socket[async_id_symbol]` but it might be stale.
[I'm not completely awake yet, but] you might want to test `socket[async_id_symbol] != socket._handle.getAsyncId()`